src/lib/dai.c

```diff
@@ -145,15 +189,23 @@ struct dai *dai_get(uint32_t type, uint32_t index, uint32_t flags)
 	if (!dti)
 		return NULL; /* type not found */
 
+	/* load possible driver from zephyr side */
+#ifdef __ZEPHYR__
+	d = dai_get_zephyr(dti, type, index);
+	if (d)
+		return d;
```

Comment:
What if dai_get_zephyr returns NULL? Currently the flow will go through non-zephyr dai creation. I think we rather should return NULL here.

Reply:
the problem is that in this transient period we have drivers both in zephyr and xtos side. So even if ZEPHYR is defined, we might have only like ssp and alh in zephyr. So the dai_get_zephyr will check the type and not do anything if driver is noton zephyr side.